pkg/scheduler/framework/parallelize/parallelism.go

```diff
@@ -53,13 +54,31 @@ func chunkSizeFor(n, parallelism int) int {
 
 // Until is a wrapper around workqueue.ParallelizeUntil to use in scheduling algorithms.
 // A given operation will be a label that is recorded in the goroutine metric.
-func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece workqueue.DoWorkPieceFunc, operation string) {
+func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece func(piece int) error, operation string) error {
```

Comment:
Or we could add another function `UntilWithErrorHandling`?

It would be good to split the refactoring into its own PR too.